### PR TITLE
GH-43357: [R] Fix some lints

### PR DESCRIPTION
**Pull request by etiennebacher**
### Rationale for this change

I'm building a new package to find and automatically fix lints in R code: [`flint`](https://flint.etiennebacher.com/).

I'm using real-life, large packages to check its performance and `arrow` is one of them. Since I already test on this, there's no additional cost for me in proposing those changes.

FYI, those changes were generated with `flint::fix_package(exclude_linters = "outer_negation")` (I excluded this linter because of https://github.com/etiennebacher/flint/issues/23).

### What changes are included in this PR?

There are many changes but most of them are trivial:
- replace `any(is.na())` by `anyNA()`
- add a leading 0 to decimal (e.g `.1` -> `0.1`)
- use `expect_named()` in tests
etc.

`flint` is quite new and linter rules are necessarily a bit opinionated, so some of those changes might not be to your taste.

### Are these changes tested?

They don't change any functionality.

### Are there any user-facing changes?

No.

* GitHub Issue: #43357

### CODE REVIEW

File: r/tests/testthat/test-dplyr-funcs-conditional.R [hunk 1]
```diff
@@ -30,7 +30,7 @@ test_that("if_else and ifelse", {
   compare_dplyr_binding(
     .input %>%
       mutate(
-        y = if_else(int > 5, 1, 0),
+        y = as.integer(int > 5),
```
Comment (thisisnic) [inline r/tests/testthat/test-dplyr-funcs-conditional.R:33]: The changes in these tests are incorrect as they completely change what is being tested here

File: r/R/arrow-tabular.R [hunk 2]
```diff
@@ -48,13 +48,13 @@ ArrowTabular <- R6Class("ArrowTabular",
     SortIndices = function(names, descending = FALSE) {
       assert_that(is.character(names))
       assert_that(length(names) > 0)
-      assert_that(!any(is.na(names)))
+      assert_that(!anyNA(names))
```
Comment (thisisnic) [inline r/R/arrow-tabular.R:51]: Hadn't heard of `anyNA()` before, nice.

**Review by thisisnic — CHANGES_REQUESTED** [review]
Hey, thanks for stopping by with these. I like the `any(is.na(...` -> `anyNA()` stuff, and equally updating the `expect_identical(names(...` to `expect_named()`, but (as general feedback), there is too much automated for me here, as some of the changes proposed break our tests and result in them testing something else entirely. (e.g. changes in [r/tests/testthat/test-dplyr-funcs-conditional.R](https://github.com/apache/arrow/pull/43338/files#diff-c63a873ac0b560d2ca7229ac1df2628df8396b0cc51fc247b71a2499c492fe3a))

**Comment by etiennebacher** [comment]
Hi, thanks for the feedback. I have reverted the changes on the `ifelse/if_else` calls as well as on the line where there was a `# nolint` (I will add a feature in `flint` to ignore those as well). 

I checked the other changes in the tests and they are only about adding a trailing `0` for decimals and using `expect_named()`.

**Comment by kou** [comment]
Could you open a new issue for this?
FYI: Our MINOR definition: https://github.com/apache/arrow/blob/main/CONTRIBUTING.md#Minor-Fixes

**Comment by etiennebacher** [comment]
I have opened #43357 for this, but I must say I don't understand the point of doing so. I get that you have some standards and clear procedures to follow to ensure consistency for thousands of external contributions, but in this case I don't see why opening an issue afterwards is useful.

**Review by thisisnic — APPROVED** [review]
@etiennebacher Thanks for taking the time to open that issue, and thanks again for your contribution here.

**Comment by etiennebacher** [comment]
Hi there, just wondering if there's anything blocking this PR. I would simply like to avoid potential merge conflicts later on.

**Pull request merged**